src/App.vue

```diff
@@ -4,22 +4,36 @@ import Login from './components/Login.vue'
 import Navbar from './components/Navbar.vue'
 import { useUserDetails } from './composables/useUserDetails'
 
-const user = useUserDetails()
-const loggedIn = computed(() => user.value?.role === 'Administrator')
+const { state, isLoading, isReady } = useUserDetails()
+const loggedIn = computed(() => state.value?.role === 'Administrator')
 </script>
 
 <template>
 	<div class="root">
-		<div v-if="loggedIn">
-			<Navbar />
-			<RouterView />
+		<Teleport to="body" v-if="isLoading">
```

Comment:
It's kinda strange to have the loader be a modal, imo. We could just keep this as a div

Reply:
I think it looks ~~good~~ acceptable like this personally. I guess the other way to do it would be to just not render anything until the content is ready. Or maybe just have a placeholder loading screen.